common/src/main/java/com/wynntils/models/skillpoint/SkillPointModel.java

```diff
@@ -0,0 +1,386 @@
+
+    public SkillPointModel() {
+        super(List.of());
+    }
+
+    public boolean hasLoadout(String name) {
+        return skillPointLoadouts.get().containsKey(name);
+    }
+
+    public void saveSkillPoints(String name, int[] skillPoints) {
+        SavableSkillPointSet assignedSkillPointSet = new SavableSkillPointSet(skillPoints);
+        skillPointLoadouts.get().put(name, assignedSkillPointSet);
+        WynntilsMod.info("Saved skill point loadout: " + name + " " + assignedSkillPointSet);
+    }
+
+    /**
+     * Saves the current assigned skill points to the loadout list.
+     * @param name The name of the loadout to save.
+     */
+    public void saveCurrentSkillPoints(String name) {
+        saveSkillPoints(name, new int[] {
+            getAssignedSkillPoints(Skill.STRENGTH),
+            getAssignedSkillPoints(Skill.DEXTERITY),
+            getAssignedSkillPoints(Skill.INTELLIGENCE),
+            getAssignedSkillPoints(Skill.DEFENCE),
+            getAssignedSkillPoints(Skill.AGILITY)
+        });
+    }
+
+    /**
+     * Saves the current equipped gear and provided skill points.
+     * @param name
+     * @param skillPoints
+     */
+    public void saveBuild(String name, int[] skillPoints) {
+        List<String> armourNames = new ArrayList<>();
+        McUtils.inventory().armor.stream()
+                .filter(x -> !x.isEmpty())
+                .map(x -> x.getHoverName().getString())
+                .forEach(armourNames::add);
+        Collections.reverse(armourNames); // helmet to boots order
+
+        List<String> accessoryNames = new ArrayList<>();
+        for (int i : ACCESSORY_SLOTS) {
+            ItemStack itemStack = McUtils.inventory().getItem(i);
+            if (!itemStack.isEmpty() && !itemStack.getHoverName().getString().equals("§7Accessory Slot")) {
+                accessoryNames.add(itemStack.getHoverName().getString());
+            }
+        }
+
+        SavableSkillPointSet assignedSkillPointSet = new SavableSkillPointSet(skillPoints, armourNames, accessoryNames);
+        skillPointLoadouts.get().put(name, assignedSkillPointSet);
+        WynntilsMod.info("Saved skill point build: " + name + " " + assignedSkillPointSet);
+    }
+
+    public void saveCurrentBuild(String name) {
+        saveBuild(name, new int[] {
+            getAssignedSkillPoints(Skill.STRENGTH),
+            getAssignedSkillPoints(Skill.DEXTERITY),
+            getAssignedSkillPoints(Skill.INTELLIGENCE),
+            getAssignedSkillPoints(Skill.DEFENCE),
+            getAssignedSkillPoints(Skill.AGILITY)
+        });
+    }
+
+    public void loadLoadout(String name) {
+        McUtils.closeBackgroundContainer();
+
+        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Loading Skill Point Loadout Query")
+                .onError(msg -> WynntilsMod.warn("Failed to load skill point loadout: " + msg))
+                .then(QueryStep.useItemInHotbar(CharacterModel.CHARACTER_INFO_SLOT - 1)
```

Comment:
I don't think we need the -1 anywhere else?

Reply:
Oh, I see now! I think you confused this variable. This variable is used for getting the player info item from the character menu, not to open the character info menu. So just introduce a new field with the correct index.
![image](https://github.com/Wynntils/Artemis/assets/49001742/d1c56b59-db64-467a-8eea-03ecfc2ef321)